storage/src/main/java/fk/prof/storage/S3AsyncStorage.java

```diff
@@ -113,22 +81,32 @@ private void initialize() {
         S3ObjectPath objectPath = new S3ObjectPath(prefixPath);
         return CompletableFuture.supplyAsync(() -> {
             try {
+                long startTime = System.currentTimeMillis();
                 Set<String> allObjects = new HashSet<>();
-                ObjectListing objects;
-                if (recursive) {
-                    objects = client.listObjects(new ListObjectsRequest().withBucketName(objectPath.bucket).withPrefix(objectPath.fileName));
-                } else {
-                    objects = client.listObjects(new ListObjectsRequest().withBucketName(objectPath.bucket).withDelimiter(DELIMITER).withPrefix(objectPath.fileName));
-                }
-
+                ObjectListing objects = null;
+                ListObjectsRequest request = new ListObjectsRequest().withBucketName(objectPath.bucket).withPrefix(objectPath.fileName);
                 do {
+                    if(!recursive) {
```

Comment:
Is this required on every iteration. If this is carried forward in next request just like withBucketName and withPrefix, then lets move it outside the loop